Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +211,193 @@ protected override bool IsValidController(IMixedRealityController controller)
+            MixedRealityPose pinkyKnuckle;
+            MixedRealityPose wrist;
+
+            if (jointedHand.TryGetJoint(TrackedHandJoint.IndexKnuckle, out indexKnuckle) &&
+                jointedHand.TryGetJoint(TrackedHandJoint.PinkyKnuckle, out pinkyKnuckle) &&
+                jointedHand.TryGetJoint(TrackedHandJoint.Wrist, out wrist))
+            {
+                handPlane = new Plane(indexKnuckle.Position, pinkyKnuckle.Position, wrist.Position);
+                activationPoint = handPlane.ClosestPointOnPlane(GenerateActivationPoint(jointedHand));
+                return true;
+            }
+            else // Otherwise, set the activation point and plane to default values and return false
+            {
+                handPlane = new Plane();
+                activationPoint = Vector3.zero;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// This function attempts to generate an activation point based on what the currently-selected safe zone is.
+        /// The activation point is a Vector3 that represents the "Center" point of the area that a user needs to gaze at to
+        /// activate the attached 
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <returns></returns>
+        private Vector3 GenerateActivationPoint(IMixedRealityHand jointedHand)
+        {
+            MixedRealityPose referenceJoint1;
+            MixedRealityPose referenceJoint2;
+
+            switch (SafeZone)
+            {
+                case SolverSafeZone.AboveFingerTips:
+                    if (!jointedHand.TryGetJoint(TrackedHandJoint.MiddleTip, out referenceJoint1) ||
+                        !jointedHand.TryGetJoint(TrackedHandJoint.RingTip, out referenceJoint2))
+                    { 
+                        return Vector3.zero;
+                    }
+                    break;
+                case SolverSafeZone.BelowWrist:
+                    return jointedHand.TryGetJoint(TrackedHandJoint.Wrist, out referenceJoint1) ? referenceJoint1.Position : Vector3.zero;
+
+                case SolverSafeZone.RadialSide:
+                    if (!jointedHand.TryGetJoint(TrackedHandJoint.IndexKnuckle, out referenceJoint1) ||
+                        !jointedHand.TryGetJoint(TrackedHandJoint.ThumbProximalJoint, out referenceJoint2))
+                    {
+                        return Vector3.zero;
+                    }
+                    break;
+                case SolverSafeZone.UlnarSide:
+                default:
+                    if (!jointedHand.TryGetJoint(TrackedHandJoint.Wrist, out referenceJoint1) ||
+                        !jointedHand.TryGetJoint(TrackedHandJoint.PinkyKnuckle, out referenceJoint2))
+                    {
+                        return Vector3.zero;
+                    }
+                    break;                       
+            }
+
+            return Vector3.Lerp(referenceJoint1.Position, referenceJoint2.Position, .5f);
```

Comment:
ah, reading through, it seems like if the result is below the wrist, you just return the wrist joint. Otherwise, lerp in between different joints based on the solver safe zone. It might be a little bit more readable to just have the switch statement pick the reference joint ids, something like:

```
switch (SafeZone)
{
    case SolverSafeZone.AboveFingerTips:
        jointId1 = TrackedHandJoint.MiddleTip;
        jointId2 = TrackedHandJoint.RingTip
   ....
}
if (!jointedHand.TryGetJoint(jointId1, out referenceJoint1) ||
   !jointedHand.TryGetJoint(jointId2, out referenceJoint2))
   {
       Debug.LogWarning("couldn't find joints");
       return Vector3.zero;
    }
return return Vector3.Lerp(referenceJoint1.Position, referenceJoint2.Position, .5f);
```

Reply:
Yeah that's way more readable; I was struggling with how to make this more sane and this really helped. Thanks Julia!